src/js/other-websites/inpage_menu.js

```diff
@@ -233,20 +354,46 @@ const buildContent = {
       );
 
       const maskLists = document.querySelectorAll(".fx-relay-menu-masks-list");
-      const masks = await getMasks();
+      masks = await getMasks();
+
+      const currentPageHostName = await browser.runtime.sendMessage({
+        method: "getCurrentPageHostname",
+      });
 
       maskLists?.forEach(async (maskList) => {
         // Set Mask List label names
-
-        const label = maskList.querySelector(".fx-relay-menu-masks-list-label");        
-
+        const label = maskList.querySelector(".fx-relay-menu-masks-list-label");
         const stringId = label.dataset.stringId;
-        
+
         label.textContent = browser.i18n.getMessage(stringId);
 
+        // If there are no masks used on the current site, we need to change the label for the other masks:
+        if (
+          !checkIfAnyMasksWereGeneratedOnCurrentWebsite(masks, currentPageHostName) &&
+          maskList.classList.contains("fx-relay-menu-masks-free-other")
+        ) {
+          // TODO: (BLOCKED) https://github.com/mozilla-l10n/fx-private-relay-add-on-l10n/pull/31
```

Comment:
`TODONE`! This is merged now, so can we update this comment and line?

Reply:
Updated to use the L10N string. Revised in d5e6221fe717faa6809d549a9e0b85259f67a063